vsphere_storage_plugin/iso9660.py

```diff
@@ -0,0 +1,100 @@
+########
+# Copyright (c) 2016-2019 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pycdlib
+import os
+import re
+from io import BytesIO
+
+from cloudify import ctx
+
+# This package imports
+from cloudify_vsphere.utils import op
+from vsphere_plugin_common import (
+    with_rawvolume_client,
+)
+from vsphere_plugin_common.constants import STORAGE_IMAGE
+
+
+def _joliet_name(name):
+    if name[0] == "/":
+        name = name[1:]
+    return "/{}".format(name[:64])
+
+
+def _name_cleanup(name):
+    return re.sub('[^A-Z0-9_]{1}', r'_', name.upper())
+
+
+def _iso_name(name):
+    if name[0] == "/":
+        name = name[1:]
+
+    name_splited = name.split('.')
+    if len(name_splited[-1]) <= 3 and len(name_splited) > 1:
+        return "/{}.{};1".format(
+            _name_cleanup("_".join(name_splited[:-1])[:8]),
+            _name_cleanup(name_splited[-1]))
+    else:
+        return "/{}.;1".format(_name_cleanup(name[:8]))
+
+
+@op
+@with_rawvolume_client
+def create(rawvolume_client, files, datacenter_name, allowed_datastores,
+           **kwargs):
+    ctx.logger.info("Creating new iso image.")
+
+    iso = pycdlib.PyCdlib()
+    iso.new(vol_ident='cidata', joliet=3, rock_ridge='1.09')
+
+    if not files:
+        files = ctx.node.properties.get('files', {})
+
+    for name in files:
```

Comment:
Should it perhaps be possible for the user to provide a list of raw files (e.g. so that they can easily add a CD with binaries on it)?

Reply:
Good idea, i will add raw_files list field - for upload some content from blueprint directory.